filebeat/input/filestream/input.go

```diff
@@ -142,10 +156,18 @@ func (inp *filestream) Run(
 		return fmt.Errorf("not file source")
 	}
 
-	log := ctx.Logger.With("path", fs.newPath).With("state-id", src.Name())
+	srcID := src.Name()
+	if inp.detailedMetricsEnabled {
+		if m, ok := inp.detailedMetrics[srcID]; ok {
+			m.cancel()
+			delete(inp.detailedMetrics, srcID)
+		}
+	}
```

Comment:
`Run` will be used from multiple go-routines. One per harvester/source. I think the removal here must be protected.

We add a new progress metrics context if `open` succeeds. If something goes wrong in `openFile`, we suddenly stop monitoring the file. Is this on purpose?

The metrics `readOffset`, `lastPublished` seem to belong to the reader, but the other metrics could be updated by the prospector or file watcher.

Reply:
Yes, it is on purpose. Why monitor the file if we cannot open it?